### PR TITLE
Clean flash management, resilient aliases and bootloaders

### PR DESCRIPTION
**Pull request by nicolas-rabault**
# PR Checklist 

### _1- PR Comments_
    *If needed, to be filled by developer*  

<br><hr/>
*WARNING : Do not edit the checklist below.*  
<hr/><br>

### _2- Developer section_

- [x] [Documentation] is up to date with new feature
- [x] [Tests] are *passed OK* (Non regression, new features & bug fixes)
- [x] [Code Quality] Remind to check if :
    * Each function has a header (description, inputs, outputs) 
    * Code is commented (particularly in hard to understand areas)
    * There are no new warnings that can be corrected
    * Commits policy is respected (constitancy commits, clear commits comments)
<hr/><br>

### _3- QA section_

- [x] [Review] tests for new features have been *reviewed*
- [x] [Changelog] is up-to-date with expected tags  
    🆕 Feature: [Feature] Description...  
    🆕 Added: [Feature] Description...  
    🆕 Changed: [Feature] Description...  
    🛠️ Fix: [Feature] Description...

### CODE REVIEW

**Review by nicolas-rabault — COMMENTED** [review]
Sounds really good!
I do have some comments about the flags...

File: engine/core/src/luos_engine.c [hunk 1]
```diff
@@ -79,7 +79,7 @@ void Luos_Loop(void)
     ll_service_t *oldest_ll_service = NULL;
     msg_t *returned_msg             = NULL;
 
-#ifndef BOOTLOADER_CONFIG
+#ifdef BOOT_APP
```
[the remark this answers is not in the document]
Comment (Salem-Luos) [reply engine/core/src/luos_engine.c:92]: ok

File: engine/HAL/STM32F0/luos_hal_config.h [hunk 2]
```diff
@@ -7,6 +7,29 @@
...
+ *
+ *             SHARED_MEMORY_ADDRESS
+ *             0x0800C000
+ *               |
+ *               |   APP_START_ADDRESS                  APP_END_ADDRESS
+ *               |   |0x0800C800                          |
+ *               |   |                                    |
+ *               v   v                                    v
+ *   +-----------+---+------------------------------------+
+ *   |   48Kb    |2Kb|           206Kb                    |
```
Comment (Salem-Luos) [inline engine/HAL/STM32F0/luos_hal_config.h:23]: all the MCU in the different familly has differente memory size I realy like this drawing but I think you should remove the app size and adapt the adress if needed

File: examples/projects/NUCLEO-F072RB/gate_serialcom/src/main.c [hunk 3]
```diff
@@ -83,7 +88,19 @@ int main(void)
     SystemClock_Config();
 
     /* USER CODE BEGIN SysInit */
+#ifdef BOOTLOADER_MODE
+    /* Copy the vector table from the Flash (mapped at the base of the application
+        load address 0x0800C800) to the base address of the SRAM at 0x20000000. */
+    for (uint32_t i = 0; i < 48; i++)
+    {
+        VectorTable[i] = *(__IO uint32_t *)(0x0800C800 + (i << 2));
```
Comment (Salem-Luos) [inline examples/projects/NUCLEO-F072RB/gate_serialcom/src/main.c]: put a define here for the adresse 0x800C8000

File: examples/projects/NUCLEO-F401RE/led/lib/Led/led.c [hunk 4]
```diff
@@ -54,4 +58,8 @@ static void Led_MsgHandler(service_t *service, msg_t *msg)
             HAL_GPIO_WritePin(LED_GPIO_Port, LED_Pin, Led_last_state);
         }
     }
+    if (msg->header.cmd == WRITE_ALIAS)
```
Comment (Salem-Luos) [inline examples/projects/NUCLEO-F401RE/led/lib/Led/led.c:61]: I don't understand how you manage that? How the command WRITE_ALIAS is treat by luos and by the service?

**Review by Salem-Luos — APPROVED** [review]

File: engine/HAL/STM32F0/luos_hal_config.h (hunk 2 above)
Comment (BiliouriV) [reply engine/HAL/STM32F0/luos_hal_config.h:23]: OK

File: examples/projects/NUCLEO-F072RB/gate_serialcom/src/main.c (hunk 3 above)
Comment (BiliouriV) [reply examples/projects/NUCLEO-F072RB/gate_serialcom/src/main.c]: OK

File: examples/projects/NUCLEO-F401RE/led/lib/Led/led.c (hunk 4 above)
Comment (BiliouriV) [reply examples/projects/NUCLEO-F401RE/led/lib/Led/led.c:61]: @nicolas-rabault

**Review by BiliouriV — APPROVED** [review]

**Pull request merged**